apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/network/http/BatchingHttpEngine.kt

```diff
@@ -195,8 +198,11 @@ class BatchingHttpEngine(
   }
 
   override fun dispose() {
-    scope.cancel()
-    dispatcher.dispose()
+    if (!disposed) {
+      delegate.dispose()
+      scope.cancel()
+      dispatcher.dispose()
```

Comment:
Unless I'm missing something, `disposed = true` is missing

Reply:
I blame copilot!

Reply:
😂